plotly/tests/test_optional/test_figure_factory.py

```diff
@@ -842,72 +842,102 @@ def test_trisurf_all_args(self):
 
         exp_trisurf_plot = {
-        'data': [
-            {
-                'facecolor': ['rgb(144.0, 94.5, 132.0)',
-                              'rgb(23.0, 190.0, 207.0)',
-                              'rgb(144.0, 94.5, 132.0)',
-                              'rgb(31.0, 119.0, 180.0)',
-                              'rgb(144.0, 94.5, 132.0)',
-                              'rgb(31.0, 119.0, 180.0)',
-                              'rgb(144.0, 94.5, 132.0)',
-                              'rgb(23.0, 190.0, 207.0)'],
-                'i': [3, 1, 1, 5, 7, 3, 5, 7],
-                'j': [1, 3, 5, 1, 3, 7, 7, 5],
-                'k': [4, 0, 4, 2, 4, 6, 4, 8],
-                'name': '',
-                'type': 'mesh3d',
-                'x': np.array([-1.,  0.,  1., -1.,  0.,  1., -1.,  0.,  1.]),
-                'y': np.array([-1., -1., -1.,  0.,  0.,  0.,  1.,  1.,  1.]),
-                'z': np.array([ 1., -0., -1., -0.,  0.,  0., -1.,  0.,  1.])
-            },
-            {
-                'line': {'color': 'rgb(50, 50, 50)', 'width': 1.5},
-                'mode': 'lines',
-                'type': 'scatter3d',
-                'x': [-1.0, 0.0, 0.0, -1.0, None, 0.0, -1.0, -1.0, 0.0, None,
-                      0.0, 1.0, 0.0, 0.0, None, 1.0, 0.0, 1.0, 1.0, None, 0.0,
-                      -1.0, 0.0, 0.0, None, -1.0, 0.0, -1.0, -1.0, None, 1.0,
-                      0.0, 0.0, 1.0, None, 0.0, 1.0, 1.0, 0.0, None],
-                'y': [0.0, -1.0, 0.0, 0.0, None, -1.0, 0.0, -1.0, -1.0, None,
-                      -1.0, 0.0, 0.0, -1.0, None, 0.0, -1.0, -1.0, 0.0, None,
-                      1.0, 0.0, 0.0, 1.0, None, 0.0, 1.0, 1.0, 0.0, None, 0.0,
-                      1.0, 0.0, 0.0, None, 1.0, 0.0, 1.0, 1.0, None],
-                'z': [-0.0, -0.0, 0.0, -0.0, None, -0.0, -0.0, 1.0, -0.0,
-                      None, -0.0, 0.0, 0.0, -0.0, None, 0.0, -0.0, -1.0, 0.0,
-                      None, 0.0, -0.0, 0.0, 0.0, None, -0.0, 0.0, -1.0, -0.0,
-                      None, 0.0, 0.0, 0.0, 0.0, None, 0.0, 0.0, 1.0, 0.0, None]
-            }
-        ],
-        'layout': {
-            'height': 800,
-            'scene': {'aspectratio': {'x': 1, 'y': 1, 'z': 1},
-            'xaxis': {'backgroundcolor': 'rgb(230, 230, 230)',
-            'gridcolor': 'rgb(255, 255, 255)',
-            'showbackground': True,
-            'zerolinecolor': 'rgb(255, 255, 255)'},
-            'yaxis': {'backgroundcolor': 'rgb(230, 230, 230)',
-            'gridcolor': 'rgb(255, 255, 255)',
-            'showbackground': True,
-            'zerolinecolor': 'rgb(255, 255, 255)'},
-            'zaxis': {'backgroundcolor': 'rgb(230, 230, 230)',
-            'gridcolor': 'rgb(255, 255, 255)',
-            'showbackground': True,
-            'zerolinecolor': 'rgb(255, 255, 255)'}},
-            'title': 'Trisurf Plot',
-            'width': 800
-        }
+            'data': [{'facecolor': np.array(
+                      [b'rgb(145.0, 96.0, 143.0)',
```

Comment:
@Kully remove the `b`'s from here and the `dtype='|S23'`. You're just being overly-strict in this test. Python 2 and 3 treat `str` differently and will create a `byte` or `unicode` string, respectively.